common/src/main/java/com/tc/util/properties/SystemPropertyCache.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Terracotta, Inc.
+ * Copyright Super iPaaS Integration LLC, an IBM Company 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tc.util.properties;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class SystemPropertyCache {
```

Comment:
Would we be better off with an `SSLSettings` class that encapsulates the property constants in one place rather than having a general purpose solution here, and still having to be careful to have static final constants elsewhere?

```java
public class SslSettings {
  public static final boolean TRUST_ALL_CERTS = Boolean.getBoolean("tc.ssl.trustAllCerts");
  public static final boolean DISABLE_HOSTNAME_VERIFIER = Boolean.getBoolean("tc.ssl.disableHostnameVerifier");
}
```